components/kyma-environment-broker/internal/broker/instance_update_test.go

```diff
@@ -41,30 +43,65 @@ func TestUpdateEndpoint_Update(t *testing.T) {
 	}
 	st := storage.NewMemoryStorage()
 	st.Instances().Insert(instance)
-	st.Operations().InsertProvisioningOperation(internal.ProvisioningOperation{
-		Operation: internal.Operation{
-			InstanceID: instanceID,
-			ProvisioningParameters: internal.ProvisioningParameters{
-				ErsContext: internal.ERSContext{
-					ServiceManager: &internal.ServiceManagerEntryDTO{
-						Credentials: internal.ServiceManagerCredentials{
-							BasicAuth: internal.ServiceManagerBasicAuth{
-								Username: "u",
-								Password: "p",
-							},
-						},
-					},
-				},
+	st.Operations().InsertProvisioningOperation(fixProvisioningOperation())
+	handler := &handler{}
+	svc := NewUpdate(st.Instances(), st.Operations(), handler, true, logrus.New())
+
+	// when
+	svc.Update(context.Background(), instanceID, domain.UpdateDetails{
+		ServiceID:       "",
+		PlanID:          TrialPlanID,
+		RawParameters:   nil,
+		PreviousValues:  domain.PreviousValues{},
+		RawContext:      json.RawMessage("{\"active\":false}"),
+		MaintenanceInfo: nil,
+	}, true)
+
+	// then
+	inst, err := st.Instances().GetByID(instanceID)
+	require.NoError(t, err)
+	// check if original ERS context is set again in the Instance entity
+	assert.NotEmpty(t, inst.Parameters.ErsContext.ServiceManager.Credentials.BasicAuth.Password)
+	// check if the handler was called
+	assert.Equal(t, &internal.ServiceManagerEntryDTO{
+		Credentials: internal.ServiceManagerCredentials{
+			BasicAuth: internal.ServiceManagerBasicAuth{
+				Username: "u",
+				Password: "p",
+			},
+		}}, handler.Instance.Parameters.ErsContext.ServiceManager)
+	assert.Equal(t, internal.ERSContext{
+		Active: ptr.Bool(false),
+	}, handler.ersContext)
+	// check if handler was called with Instance.active=true
+	require.NotNil(t, *handler.Instance.Parameters.ErsContext.Active)
```

Comment:
this comment is kinda misleading - we call update with `active: false` payload and so is handler called